lib/sidekiq/throttled/patches/super_fetch.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require "sidekiq"
+
+module Sidekiq
+  module Throttled
+    module Patches
+      module SuperFetch
+        # Retrieves job from redis.
+        #
+        # @return [Sidekiq::Throttled::UnitOfWork, nil]
+        def retrieve_work
+          work = super
+
+          if work && Throttled.throttled?(work.job)
+            Throttled.cooldown&.notify_throttled(work.queue)
+            requeue_throttled(work)
+            return nil
+          end
+
+          Throttled.cooldown&.notify_admitted(work.queue) if work
+
+          work
+        end
+
+        private
+
+        # Pushes job back to the head of the queue, so that job won't be tried
+        # immediately after it was requeued (in most cases).
+        #
+        # @note This is triggered when job is throttled. So it is same operation
+        #   Sidekiq performs upon `Sidekiq::Worker.perform_async` call.
+        #
+        # @return [void]
+        def requeue_throttled(work)
+          if work.respond_to?(:local_queue)
```

Comment:
Can we have a non-superfetch UnitOfWork when superfetch is enabled? If not, then this check is not needed as well as the other branch in this condition.

Reply:
No there isn't. SuperFetch's `retrieve_work` returns either a SuperFetch UnitOfWork, or nil, so I'll remove this condition